gnoi/reset/client_test.go

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2020 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package reset
+
+import (
+	"testing"
+
+	"github.com/google/gnxi/gnoi/reset/pb"
+)
+
+func TestStart(t *testing.T) {
+	t.Run("ResetSuccess", func(t *testing.T) {
```

Comment:
Would this be more readable with:

```
// in client.go:
type ResetError struct{
    Msg string
}

func (re *ResetError)Error() string{
	return re.Msg
}

// in here:

	tests := []struct {
		response *pb.StartResponse{}
		want *ResetError
	}{
		{
			response: &pb.StartResponse{Response: &pb.StartResponse_ResetError{&pb.ResetError{
			           FactoryOsUnsupported: true,
			           ZeroFillUnsupported:  false,
			           Other:                false,
			           Detail:               "",
                         },},},
			want: errors.NewError("Factory OS Rollback Unsupported"),
		},
...
	}
	for _, test := range tests {
		got := CheckResponse(test.response)
                 diff := cmp.Diff(test.want, got)
		if diff != "" {
			t.Errorf("CheckResponse(%s): (-want +got):\n%s", test.response, diff)
		}
	}
}
```

Reply:
I think this is resolved in new commit, I kept the initializeResponse function but I can remove it if you prefer, thank you